### ISSUE TITLE
Tasker Support

### ISSUE THREAD

**Issue by MarkAdamson** [home-assistant/home-assistant-android#232]
Many Android apps include support for [Tasker](https://tasker.joaoapps.com/), a phone automation app which supports third-party plugins, allowing events on the phone to trigger actions in the app.

This seems like a natural fit for the Home Assistant companion app. It would allow Tasker users to call services, render templates, and get entity states directly within Tasker automations. The classic example is being able to mute/pause media players when receiving a phone call.

I currently maintain the standalone [Home Assistant Plug-In for Tasker](https://github.com/MarkAdamson/home-assistant-plugin-for-tasker), but I would love to merge the functionality with the official Home Assistant app, allowing me to reduce the number of Home Assistant apps I have on my phone.

**Comment by JBassett** [comment]
Duplicate of https://github.com/home-assistant/home-assistant-android/issues/58

We currently consider this out of scope for the application since there is already a good solution.

**Comment by MarkAdamson** [comment]
That's disappointing :(

Would you perhaps reconsider? For a number of reasons:

- As I mentioned previously, many apps offer Tasker integration as a core feature - power users (I think the core audience of Home Assistant) expect and appreciate this kind of integration.
- Again, I feel the functionality it would offer is a natural fit for Home Assistant.
- There are already good solutions available for push notifications and location tracking - this wasn't seen as a reason not to offer them in the official app.
- I (and I'm sure others) would appreciate not having to have _one more_ Home Assistant app installed on my phone.
- As a lone developer, the existing solution simply doesn't have the reach and visibility of the official app. It's been out 6 months and I _still_ see people asking how to integrate Tasker/HA, or following complicated guides and handcrafting tricky API calls.
- As the creator of the existing solution, I'd be happy to integrate it into the official app myself - adding value at very little cost to yourselves.

Thanks for the compliment on my project though, it's much appreciated. Of course, I'm a huge fan of Home Assistant :)

**Comment by JBassett** [comment]
I was a little quick to close this issue.  If you are willing to add the functionality we will be more than willing to accept the PR into the application.  We could always use another contributor to the app!

**Comment by MarkAdamson** [comment]
Great news, thanks for reconsidering :)

I'd be more than happy to do it. I'll let you know when I have something!

**Comment by balloob** [comment]
Feel free to join us in #devs_mobile on Discord !

**Comment by MarkAdamson** [comment]
Thanks, I will - I expect I'll have a few questions as I get started!

**Comment by scottismyname** [comment]
Really glad to see this. Tasker support would be amazing!

**Comment by heytcass** [comment]
I just wanted to come and voice additional support for this being integrated. I would love to be able to use additional Tasker functionality through HA without having to configure in multiple places. This would solve an issue I've been trying to solve with MQTT immediately.

**Comment by fernandex00** [comment]
Really glad to see this

**Comment by BCNelson** [comment]
I'm really glad to see this. One think that I think you be extremely valuable is to add the ability to send "hidden" push notifications from the server into Tasker so that polling is not required. Enforcing all of the same limits that are currently on visible notifications. 

This is currently possible with Tasker but a little hacky as it requires a visual interruption to and parsing the notification.

**Comment by stainlessray** [comment]
I #cosign this request. godspeed to all who help. Please feel free to reach out for testers. I would love to test and feedback to contribute.

**Comment by robbiet480** [comment]
Hey @MarkAdamson, just checking in on progress. Feel free to join us in #devs_apps on Discord if you have questions.

**Comment by BCNelson** [comment]
With the new Tasker release 5.10 we can use [Taker Commands](https://tasker.joaoapps.com/commandsystem.html) to communicate with tasker rather than building a full integration.

**Comment by Bruceforce** [comment]
I don't think this should be closed since it's not resolved unless I've missed something.

**Comment by dshokouhi** [comment]
> I don't think this should be closed since it's not resolved unless I've missed something.

our stale bot does a check in every now and then and this enhancement has not seen any movement at all.

The app has also enhanced features around intents so tasker can actually pick up on some data without needing to create a plugin.  Not sure if that is what you guys are after but another thing that can be done in the meantime.

**Comment by Bruceforce** [comment]
Thanks for your reply. A plugin would just be nice to make it easily configurable in tasks like "If WiFi disconnected turn off the lights xy" without having to search for intents oder device IDs first.

I currently use the HA API to control things with Tasker. But this always involves some extra steps like searching for a device ID in HA. So if I need to do the same for the HA Android app I see no point in sending an intent to the app versus sending an http post request directly to the api.

**Comment by dshokouhi** [comment]
> Thanks for your reply. A plugin would just be nice to make it easily configurable in tasks like "If WiFi disconnected turn off the lights xy" without having to search for intents oder device IDs first.
> 
> I currently use the HA API to control things with Tasker. But this always involves some extra steps like searching for a device ID in HA. So if I need to do the same for the HA Android app I see no point in sending an intent to the app versus sending an http post request directly to the API.
> 
> This isn't meant as offense. I'm just illustrating my use case.



No problem I know you gave wifi as just an example however the app has had a wifi sensor so you can do those automations directly in HA once you enable the wifi connection sensor.  No need to involve tasker, the state updates immediately.

You also don't need to search for intents, you can create your own custom intent that you send from HA and in tasker you just listen for that custom intent and activate your tasks based on that.  A custom intent can be any string like `my.custom.intent`.  Vice versa is also possible.

The HA app can also make things easier and in some cases remove the need for tasker so if you see a certain feature that is being done in tasker and it might make sense for the app to do it we can consider that as a separate feature request.  I wouldn't say the app is a replacement but in some cases we can make things easier.  Like TTS notifications and sending a lot of sensor data back home.

**Comment by mshuflin** [comment]
I would also like to see this implemented

**Comment by ajayjohn** [comment]
Same here.
